tests/tests.py

```diff
@@ -396,3 +405,13 @@ def test_expiry_is_present(self):
             response.data['expiry'],
             DateTimeField().to_representation(AuthToken.objects.first().expiry)
         )
+
+    def test_authenticate_credentials_handle_database_error_on_renew_token(self):
+        instance, token = AuthToken.objects.create(user=self.user)
+        with override_settings(REST_KNOX=auto_refresh_knox):
+            reload_module(auth)
+            token_auth = TokenAuthentication()
+            with mock.patch.object(token_auth, 'renew_token') as m:
+                m.side_effect = DatabaseError()
+                with self.assertRaises(AuthenticationFailed):
+                    token_auth.authenticate_credentials(token.encode('utf-8'))
```

Comment:
@xrmx  Thank you very much for your contribution. The scenario how you found this bug is something many could experience, however I feel your unit test is rather abstract (name and scenario). I would aim for something like test_handles_concurrent_logout_calls_as_bla_bla and try to rebuilt exactly what you experienced.

Reply:
@belugame i can rename it but i cannot replicate the exact race condition deterministically inside an unit test thus the renew_token DatabaseError mock which is what i got in real life.
Please backport master 4.1.0 commit to develop so than i can rebase cleanly.

Reply:
@xrmx if you provide a PR with the 4.1.0 commits in develop as I suggested before, myself or @belugame would gladly merge that.

Reply:
Updated test name and added comments to describe the issue